core/src/main/java/org/apache/iceberg/FilteredManifest.java

```diff
@@ -95,60 +95,31 @@ public FilteredManifest caseSensitive(boolean isCaseSensitive) {
       Evaluator evaluator = evaluator();
       InclusiveMetricsEvaluator metricsEvaluator = metricsEvaluator();
 
-      return CloseableIterable.filter(reader.entries(projection(fileSchema, columns, caseSensitive)),
+      // ensure stats columns are present for metrics evaluation
+      Collection<String> projectColumns = withStatsColumns(columns);
+
+      CloseableIterable<ManifestEntry> entries = CloseableIterable.filter(
+          reader.entries(projection(fileSchema, projectColumns, caseSensitive)),
           entry -> entry != null &&
               evaluator.eval(entry.file().partition()) &&
               metricsEvaluator.eval(entry.file()));
 
+      boolean dropStats = dropColumnStats(columns);
+      return CloseableIterable.transform(entries, e -> dropStats ? e.copyWithoutStats() : e.copy());
```

Comment:
Overall, I like the idea behind the changes to this file. It's a good idea to simply filter `allEntries()` to produce `liveEntries()`. But the change to copy `ManifestEntries` is going to hurt job planning performance. `ManifestReader` and `FilteredManifest` purposely do not copy entries to avoid the additional overhead while scanning. `ManifestEntry` is internal so anything that handles them should know when to make a defensive copy or not.

The public part of this class is the `Iterable<DataFile>`, which makes a defensive copy of all matching data files. That iterator should still make a copy, while the entries methods should not.